src/modality-specific-files/eye-tracking.md

```diff
@@ -0,0 +1,153 @@
+-->
+{{ MACROS___make_filetree_example(
+
+   {
+   "sub-<label>": {
+      "<datatype>": {
+         "<matches>_eyetrack.json": "",
+         "<matches>_eyetrack.tsv.gz": "",
+         },
+      }
+   }
+
+) }}
+
+Eye-tracking data MUST be stored in the main data recording modality or
+ `<datatype>` directory (for example `func` when combined with fMRI, or
+ `beh` when combined with behavioral measures).
+
+In the template filenames, the `<matches>` part corresponds to task filename
+before the suffix. For example, when fMRI is considered the main modality with
+files such as `sub-01_ses-1_task-pRF_run-1_bold.nii.gz`, `<matches>` would
+then correspond to `sub-01_ses-1_task-pRF_run-1`.
+
+### Tabular file (`*_eyetrack.tsv.gz`)
+
+The eye-tracking data must be saved in `.tsv.gz` format.
+The tabular files consist of one row per sample and a set of REQUIRED and
+OPTIONAL columns.
+
+<!--
+This block generates a columns table.
+The definitions of these fields can be found in
+  src/schema/rules/tabular_data/eyetrack.yaml
+and a guide for using macros can be found at
+ https://github.com/bids-standard/bids-specification/blob/master/macros_doc.md
+-->
+{{ MACROS___make_columns_table("eyetrack.EyeTracking") }}
+
+Throughout BIDS you can indicate missing values with `n/a` (for "not
+available").
+
+We encourage users to put the eye-tracking raw data within the
+`/sourcedata` directory (for example put eye-tracker proprietary files before
+conversion to `tsv.gz` format in the sourcedata directory).
+
+### Sidecar JSON document (`*_eyetrack.json`)
+
+<!-- This block generates a metadata table.
+These tables are defined in
+ src/schema/rules/sidecars
+The definitions of the fields specified in these tables may be found in
+ src/schema/objects/metadata.yaml
+A guide for using macros can be found at
+ https://github.com/bids-standard/bids-specification/blob/master/macros_doc.md
+-->
+{{ MACROS___make_sidecar_table("eyetrack.EyeTrackingRequired") }}
+
+Note that `ScreenSize`, `ScreenResolution` and `ScreenDistance` are REQUIRED as
+they are considered essential in eye-tracking data analysis.
+
+**Specific eye-tracking fields SHOULD be present:**
+
+<!-- This block generates a metadata table.
+These tables are defined in
+ src/schema/rules/sidecars
+The definitions of the fields specified in these tables may be found in
+ src/schema/objects/metadata.yaml
+A guide for using macros can be found at
+ https://github.com/bids-standard/bids-specification/blob/master/macros_doc.md
+-->
+{{ MACROS___make_sidecar_table("eyetrack.EyeTrackingRecommended") }}
+
+**Specific eye-tracking fields MUST be present:**
+
+### Example sidecar JSON document
+
+```JSON
+{
+"Manufacturer": "SR-Research",
+"ManufacturersModelName": "EYELINK II CL v4.56 Aug 18 2010",
+"SoftwareVersion": "SREB1.10.1630 WIN32 LID:F2AE011 Mod:2017.04.21 15:19 CEST",
+"SamplingFrequency": 1000,
+"SampleCoordinateUnits": "pixel",
```

Comment:
you should somehow clarify that this is the unit for gaze position. You also need a field for the unit for pupil size.

Reply:
ping

Reply:
I had understood that this unit applied to all the other values that appeared in the sidecar json (like ScreenAOIDefinition) or the datafile, but making it explicit could help.

Reply:
@dcnieho it is a good point, `SampleCoordinateUnit` is for the gaze position... we could clarify this in the description.
I feel we should change it to `GazeCoordinateUnit`, and `SampleCoordinateSystem` to `GazeCoordinateSystem` We could then add a `PupilCoordinateUnit`, but I thought that it was a arbitrary unit... @dcnieho what would be the unit? 

@Remi-Gau if we would make these changes, comes a tricky situation, as pupil size is not a required columns of the tabular file, can we do something to make required the unit of the pupil only if at least one column pupil is present in the tabular file. 
Thus would it become a "required" field or a "recommended" field ?

Reply:
Yes, sample is a vague word, gaze would be better. What the unit of pupil is depends on the eye tracker, for the eyelink its some form of area measure in what is effectively an arbitrary unit (it may well be a number of pixels, but i'm not so sure its that simple). With some prying, you can also get pupil diameters out of an eyelink, those are in pixels.
For a tobii, you get pupil diameter in mm. for an SMI pupil diameter in pixels, at least for the remote systems.

So there are really two factors to encode here:
- type: area, diameter
- unit (pixels, mm, a.u.)

It should be noted that there is at least one eye tracker out there that only gives you gaze, no pupil size. Thats the pupil invisible, which is not an uncommon eye tracker already, and looks poised to get more common. So indeed, pupil size shouldn't be a required column. I realize you are discussion something else there with Remi, but wanted to make this related point.

Reply:
Thanks, we will do something like this as soon as i got an answer on how to proceed with "required"/"required" story...